executor/script.go

```diff
@@ -169,23 +179,54 @@ func RunScript(script Script, timeout int) ExecutionResult {
 					Help:   script.Help,
 				},
 			},
+			TotalExecTime: execTotalMs,
+		}
+	}
+
+	log.Debugf("Running %s with timeout of %v", script.Path, (time.Duration(int(timeout)) * time.Second))
+
+	// The following block of code is used instead of CommandContext as the context
+	// doesn't terminate sub-processes.
+	// See: https://github.com/golang/go/issues/22485
+
+	cmd := exec.Command("/bin/bash", "-c", script.Path)
+	cmd.SysProcAttr = &syscall.SysProcAttr{Setpgid: true}
+	killChanRes := make(chan ExecutionResult, 1)
+
+	time.AfterFunc(time.Duration(int(timeout))*time.Second, func() {
+		syscall.Kill(-cmd.Process.Pid, syscall.SIGKILL)
+		duration := time.Since(execStart)
+		execTotalMs := duration.Nanoseconds() / 1000000
+		killChanRes <- ExecutionResult{
```

Comment:
I suggest you fill this and send the message to the channel before killing the process. See my point below.